requirements/customer/scheduling-algorithm.tex

```diff
@@ -63,4 +63,28 @@ \section{Scheduling Algorithm}
   By default, the scheduling algorithm will \textbf{not} pause running jobs to free up resources to run non-urgent jobs.
   However, this behavior changes once the effective priority $p$ of a non-urgent job reaches a certain threshold:
   once the threshold is reached the scheduling algorithm will begin to pause non-urgent running jobs with lower effective priority $p$ if possible and necessary to run the job with an effective priority $p$ above the threshold.
-\end{enumerate}
\ No newline at end of file
+\end{enumerate}
+\begin{figure}[H]
```

Comment:
Yea you are right. I just googled how it is done and this is the fist thing that came up. How exactly should i add references to the graphics.  Sorry I'm kind of a beginner with tex.

Reply:
1. Add `\label{\<name\>}` to the figure.
2. Add `\ref{\<name\>}` somewhere else to add a number corresponding to the figure/table/section that you labeled.

So something like:
A graphical representation of the scheduling algorithm is shown in figure `\ref{\<name\>}`

Reply:
I don't get how this fixes the problem with the graphics ending in the following section.

Reply:
If you add references graphics do not have to be in the same section.
The problem with graphics being in the next section would normally be that there is confusion as to which section the graphics belong to.
But if you refer to your graphics in the text there will be no confusion.

Reply:
But i tried it, and it didn't .

Reply:
I think there is a misunderstanding.
For LaTeX documents the styling convention is that graphics do **not** have to be in the same section as the text they belong to.
This is because you are usually not starting a new page for each section.
The alternative would be to use LaTeX chapters at the highest level (and then section, subsection, subsubsection under that).
With LaTeX chapters there would be a new page for each chapter and graphics would stay in the chapter they are defined in.
However, for short documents like ours this would add a relatively high amount of unused blank space.

Reply:
Ok, but we want then not going in the next section, and the using \ref{} doesn't do the job for me. They still shift to the next one.

Reply:
Okay.
But if we want to do that we should also start a new page for each section to keep the style consistent.
https://github.com/DistributedTaskScheduling/documentation/pull/48 is actually doing just that.